FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -175,6 +175,12 @@ class Flagsmith constructor(
         retrofit.getIdentityFlagsAndTraits(identity).enqueueWithResult(defaults = null, result = result)
             .also { lastUsedIdentity = identity }
 
+    fun getIdentityFlags(identity: String, traits: List<Trait>, result: (Result<ArrayList<Flag>>) -> Unit) {
```

Comment:
I think better rename `getIdentityFlags` to `setAndGetIdentityFlags`

Reply:
@devapro the idea here was to keep things consistent with a lot of our other SDKs which use the `getIdentityFlags` method name, however, I have some other feedback which I'm compiling the related PR in the ios client [here](https://github.com/Flagsmith/flagsmith-ios-client/pull/59).

Reply:
For me, the function name `getIdentityFlags` is a bit confusing, because it doesn't match what this method does.
>Get flags for an identity and set traits in the same call
Even in the code of this method, we can see the call of `postTraits`

An option that you propose in iOS PR (Add an optional traits argument to the existing `getFeatureFlags` method ) looks good to me. However, I think we don't need to raise an exception if the identity is null because there already is a condition `if (identity != null)`, maybe we can use it and  `postTraits` only if the identity is not null.

Reply:
>However, I think we don't need to raise an exception if the identity is null because there already is a condition if (identity != null), maybe we can use it and postTraits only if the identity is not null.

In my opinion it makes sense to raise an exception here because it is an invalid set of arguments. I'm not sure what the benefit is of not raising an exception in this case? I think that, without raising the exception, we may end up with confusing behaviour (since the `traits` argument would be ignored in situations where the identity is not provided).